src/diffusers/pipelines/kandinsky2_2/pipeline_kandinsky2_2_prior_emb2emb.py

```diff
@@ -156,6 +156,8 @@ def interpolate(
         self,
         images_and_prompts: List[Union[str, PIL.Image.Image, torch.FloatTensor]],
         weights: List[float],
+        image: Union[torch.Tensor, List[torch.Tensor], PIL.Image.Image, List[PIL.Image.Image]],
+        strength: float = 0.3,
```

Comment:
```suggestion
        images_and_prompts: List[Union[str, PIL.Image.Image, torch.FloatTensor]],
        weights: List[float],
        image: Union[torch.Tensor, List[torch.Tensor], PIL.Image.Image, List[PIL.Image.Image]],
        strength: float = 0.3,
```

It's a bit confusing that we have both `images_and_prompts` and `image` as inputs I think. Also could it be possible to provide multiple images if there are multiple prompts?  

Should we maybe refactor the function here and instead have the following inputs:

```suggestion
        prompts: Optional[Union[List[str], str]],
        prompt_images: Optional[Union[List[PIL.Image], PIL.Image]],
        images: Optional[Union[List[PIL.Image], PIL.Image]],
        weights: List[float],
        image: Union[torch.Tensor, List[torch.Tensor], PIL.Image.Image, List[PIL.Image.Image]],
        strength: float = 0.3,
```

and then we deprecate `images_and_prompts`? (and we maybe do the same for Kandinsky 2.1?)

Reply:
I'm a bit worried that `images_and_prompts` is becoming a bit too messy. Wdyt @yiyixuxu ?

Reply:
@patrickvonplaten 

Are we trying to replace `images_and_prompts` with `prompts` and `prompt_images`? what is the `images` argument? Not sure this simplifies things because we also have this `weights` argument, which should be a list that has the same length as `images_and_prompts` so that we use the first weight for the first thing in `images_and_prompts`......

I agree it's really confusing to have both `images_and_prompts` and `image` arguments, I think maybe we can rename `images_and_prompts` to just `prompts`? I understand that when we talk about "prompts" we usually refer to a text. However, this is not true for this pipeline, both images and texts passed as `images_and_prompts` will be converted into the CLIP image embedding and act the same role in the diffusion process as a "prompt" does. so if we only have `prompts` and `images`, it is much easier to understand which argument does what - we can then make it clear in docstring that you can pass prompt in both image and text form.

Reply:
I would maybe rename it to "conditions" - think "images_and_prompts" is not a waterproof name going forward